plugins/org.python.pydev.debug/src/org/python/pydev/debug/ui/PathMappingsPrefsPage.java

```diff
@@ -0,0 +1,43 @@
+package org.python.pydev.debug.ui;
+
+import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.python.pydev.plugin.PydevPlugin;
+import org.python.pydev.plugin.preferences.PyDevEditorPreferences;
+import org.python.pydev.shared_ui.field_editors.MultiStringFieldEditor;
+
+/**
+ * Preferences for the locations that should be translated -- used when the debugger is not able
+ * to find some path aa the client, so, the user is asked for the location and the answer is
+ * kept in the preferences in the format:
+ * 
+ * path asked, new path -- means that a request for the "path asked" should return the "new path"
+ * path asked, DONTASK -- means that if some request for that file was asked it should silently ignore it
+ */
+public class PathMappingsPrefsPage extends FieldEditorPreferencePage implements IWorkbenchPreferencePage {
+
+    /**
+     * Initializer sets the preference store
+     */
+    public PathMappingsPrefsPage() {
+        super("Path Mapping", GRID);
+        setPreferenceStore(PydevPlugin.getDefault().getPreferenceStore());
+    }
+
+    @Override
+    public void init(IWorkbench workbench) {
+    }
+
+    /**
+     * Creates the editors
+     */
+    @Override
+    protected void createFieldEditors() {
+        Composite p = getFieldEditorParent();
+        addField(new MultiStringFieldEditor(PyDevEditorPreferences.PATHS_FROM_ECLIPSE_TO_PYTHON,
```

Comment:
Instead of using `MultiStringFieldEditor` directly, you may need to do something a bit more custom to validate that the entry is indeed correct and show parsing errors to the user.

We should create a `JsonFieldEditor` which would allow us to do this (it may come in handy for other editors in the future too).

Reply:
Note: the field (JSonFieldEditor) should wrap a `StyledText` (which should allow us to colorize the contents properly later on).